test/Interop/Cxx/stdlib/use-std-function.swift

```diff
@@ -9,25 +9,52 @@
 
 import StdlibUnittest
 import StdFunction
+import CxxStdlib
 
 var StdFunctionTestSuite = TestSuite("StdFunction")
 
-StdFunctionTestSuite.test("init empty") {
+StdFunctionTestSuite.test("FunctionIntToInt.init()") {
   let f = FunctionIntToInt()
   expectTrue(isEmptyFunction(f))
 
   let copied = f
   expectTrue(isEmptyFunction(copied))
 }
 
-StdFunctionTestSuite.test("call") {
+StdFunctionTestSuite.test("FunctionIntToInt.callAsFunction") {
   let f = getIdentityFunction()
   expectEqual(123, f(123))
 }
 
-StdFunctionTestSuite.test("retrieve and pass back as parameter") {
+StdFunctionTestSuite.test("FunctionIntToInt retrieve and pass back as parameter") {
   let res = invokeFunction(getIdentityFunction(), 456)
   expectEqual(456, res)
 }
 
+#if !os(Windows) // FIXME: rdar://103979602
+StdFunctionTestSuite.test("FunctionIntToInt init from closure and call") {
+  let cClosure: @convention(c) (Int32) -> Int32 = { $0 + 1 }
+  let f = FunctionIntToInt(cClosure)
```

Comment:
Do we want to have a test case where the closure actually captures some local state? Or is it not yet supported?

Reply:
That's not supported yet. `@convention(c)` closures (aka the ones that can be bridged to C function pointers) cannot capture context in Swift. In the future we'll need to add support for converting `@convention(thick)` closures to `std::function`, but it requires more work.